onnxruntime/core/providers/dml/OperatorAuthorHelper/MLOperatorAuthorHelper.h

```diff
@@ -13,6 +13,8 @@
 #define ML_CHECK_BOOL(x) THROW_HR_IF(E_INVALIDARG, !(x))
 #endif
 
+#include <optional>
+
```

Comment:
Move up to by top. General headers like anything in `std` should be included before app-specific definitions like `ML_CHECK_BOOL`.